pkg/provisioner/ironic/ironic.go

```diff
@@ -250,6 +269,23 @@ func (p *ironicProvisioner) findExistingHost() (ironicNode *nodes.Node, err erro
 		}
 	}
 
+	// Try to load the node by port
+	p.log.Info("looking for existing node by MAC", "MAC", p.host.Spec.BootMACAddress)
+	allPorts, err := p.listAllPorts(p.host.Spec.BootMACAddress)
```

Comment:
Minor point, but maybe all this block could be moved into a `checkExistingNodeByMac` method, and maybe `listAllPorts` could be merged within (in any case it's using the `host` member, that could be accessed directly from within the dunc